tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -698,15 +724,33 @@ def from_generator(generator, output_types, output_shapes=None, args=None):
     >>> list(dataset.take(3).as_numpy_iterator())
     [(1, array([1])), (2, array([1, 1])), (3, array([1, 1, 1]))]
 
-    Note: The current implementation of `Dataset.from_generator()` uses
+    Use `output_spec` for more complicated cases, e.g. to specify an output
+    containing `tf.RaggedTensor`, `tf.SparseTensor` or other objects different
+    from `tf.Tensor`:
+
+    >>> def gen():
+    ...   ragged_tensor = tf.ragged.constant([[1, 2], [3]],
+    ...                                      ragged_rank=1,
+    ...                                      dtype=tf.int64)
+    ...   yield 42, ragged_tensor
+    >>>
+    >>> dataset = tf.data.Dataset.from_generator(
+    ...      gen,
+    ...      output_spec=(tf.TensorSpec(shape=(), dtype=tf.int64),
+    ...                   tf.RaggedTensorSpec(shape=(2, None), dtype=tf.int64)))
+    >>>
+    >>> list(dataset.take(1))
+    [(<tf.Tensor: shape=(), dtype=int64, numpy=42>, <tf.RaggedTensor [[1, 2], [3]]>)]
+
+    NOTE: The current implementation of `Dataset.from_generator()` uses
```

Comment:
Note: ... This will result in markdown rendering the paragraph with a different style.

Reply:
Oh, I see, thanks

Reply:
fixed